library/src/main/scala/scala/scalajs/js/WeakRef.scala

```diff
@@ -0,0 +1,23 @@
+package scala.scalajs.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+/** <span class="badge badge-ecma2021" style="float: right;">ECMAScript 2021</span>
+ *
+ * A WeakRef object lets you hold a weak reference to another object, without preventing
+ * that object from getting garbage-collected.
+ *
+ * MDN
+ *
+ * @param value
+ * @tparam T js.Object or AnyRef
```

Comment:
Interesting :) Will this make us commit to `AnyRef` being a reference for the JS GC?

Reply:
The relevant part of the spec is Step 2 of https://tc39.es/proposal-weakrefs/#sec-weak-ref-target:

> If Type(target) is not Object, throw a TypeError exception.

But there are `AnyVal`s that are guaranteed to be Objects when they reach JavaScript (any custom value class), and at the same time there are `js.Any`s that are primitives (`js.BigInt`, for example).

I don't think we should say "js.Object or AnyRef". We should let `T` be unconstrained, but document that the constructor will throw a TypeError if `Type(target)` is not Object. That is the only non-lying statement we can make.

Reply:
Updated scaladoc and added header comment